packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -143,7 +143,12 @@ class ReactShallowRenderer {
 
     this._rendering = true;
     this._element = element;
-    this._context = getMaskedContext(element.type.contextTypes, context);
+
+    if (element.type.contextType) {
+      this._context = context;
```

Comment:
This is the legacy context object. It has no relation to the new context API you're implementing.